src/collapse/dtrace.rs

```diff
@@ -1,49 +1,93 @@
-use super::Collapse;
-use fnv::FnvHashMap;
-use log::warn;
 use std::collections::VecDeque;
-use std::io;
-use std::io::prelude::*;
+use std::io::{self, prelude::*};
+use std::mem;
 
-/// Settings that change how frames are named from the incoming stack traces.
-///
-/// All options default to off.
-#[derive(Clone, Debug, Default)]
+use crossbeam::channel;
+use log::warn;
+
+use crate::collapse::{self, Collapse, Occurrences};
+
+/// Dtrace folder configuration options.
+#[derive(Clone, Debug)]
 pub struct Options {
-    /// include function offset (except leafs)
+    /// Include function offset (except leafs). Default is `false`.
     pub includeoffset: bool,
+
+    /// The number of stacks in each job sent to the threadpool (if using multiple threads).
+    /// Default is `100`.
+    pub nstacks_per_job: usize,
+
+    /// The number of threads to use. Default is the number of logical cores on your machine.
+    pub nthreads: usize,
+}
+
+impl Default for Options {
+    fn default() -> Self {
+        Self {
+            includeoffset: false,
+            nstacks_per_job: collapse::DEFAULT_NSTACKS,
+            nthreads: *collapse::DEFAULT_NTHREADS,
+        }
+    }
 }
 
 /// A stack collapser for the output of dtrace `ustrace()`.
 ///
 /// To construct one, either use `dtrace::Folder::default()` or create an [`Options`] and use
 /// `dtrace::Folder::from(options)`.
-#[derive(Default)]
 pub struct Folder {
-    /// Function entries on the stack in this entry thus far.
-    stack: VecDeque<String>,
+    /// Vector for processing java stuff
```

Comment:
This is sort of orthogonal, but it's interesting to me that you prefer fields to be in alphabetic order. I usually like them to be "semantically grouped". They weren't really grouped that way here, but I like to have fields that are related to the same "kinds" of processing or same "step" together.

Reply:
Good to know! I'll be on the lookout for where I can do that in the future now that I know your preference.

Reply:
Not terribly important, I've just found it can help with development later when all the fields that are related to each "part" of the program are close together.